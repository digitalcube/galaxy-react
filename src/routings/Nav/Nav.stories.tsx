import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Nav, NavProps } from './Nav';
import { NavLinkItem } from './NavLinkItem';

const meta: Meta = {
  title: 'Routings/Nav',
  component: Nav,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavProps> = (args) => (
  <nav className="navbar navbar-expand-lg">
    <Nav {...args}>
      {args.children}
      <NavLinkItem to="/">Menu</NavLinkItem>
      <NavLinkItem to="/">Menu</NavLinkItem>
      <NavLinkItem to="/">Menu</NavLinkItem>
    </Nav>
  </nav>
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  children: 'item',
};
