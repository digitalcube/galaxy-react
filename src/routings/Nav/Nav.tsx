import React, { FC, PropsWithChildren, useState } from 'react';
import { Collapse, Nav as ReactstrapNav, NavbarToggler } from 'reactstrap';
export type NavProps = PropsWithChildren<{}>;
export const Nav: FC<NavProps> = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);
  return (
    <>
      <NavbarToggler onClick={toggle} />
      <Collapse isOpen={isOpen} className="navbar-collapse flex-grow-0">
        <ReactstrapNav className="ml-auto navbar-nav">{children}</ReactstrapNav>
      </Collapse>
    </>
  );
};
