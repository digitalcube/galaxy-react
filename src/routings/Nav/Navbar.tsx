import React, { FC, PropsWithChildren, CSSProperties } from 'react';

export type NavbarProps = PropsWithChildren<{
  fixed?: boolean
}>;
export const Navbar: FC<NavbarProps> = ({ children, fixed }) => {
  const style: CSSProperties = fixed ? {
    position: 'fixed',
    zIndex: 999,
    background: '#fff',
    width: '100%',
    boxShadow: '0 1px 8px rgb(69 29 57 / 10%)',
  }: {}

  return (
    <nav className="navbar navbar-expand-lg" style={style}>{children}</nav>
  );
  }
