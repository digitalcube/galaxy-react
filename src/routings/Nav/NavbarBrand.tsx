import React , {FC} from 'react'
import { NavbarBrand as ReactstrapNavbarBrand } from 'reactstrap'
import { BrandName, Logo } from '../../components'
import { useLinkTag } from '../../providers'


export type NavbarBrandProps = {
    name: BrandName;
    width?: number;
    href?: string;
}

export const NavbarBrand: FC<NavbarBrandProps> = ({name, width = 24, href = '/'}) => {
    const { tag } = useLinkTag()
    return (
        <ReactstrapNavbarBrand tag={tag} href={href}>
            <Logo name={name} width={width} />
        </ReactstrapNavbarBrand>
    )
}