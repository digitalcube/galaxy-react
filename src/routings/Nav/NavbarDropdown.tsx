import React, { FC, PropsWithChildren } from 'react';
import { UncontrolledDropdown, DropdownToggle, DropdownMenu } from 'reactstrap';
import { Avatar, AvatarProps } from '../../components/Avatar/Avatar';

export type NavbarDropdownProps = PropsWithChildren<{
  avatar: AvatarProps;
}>;

export const NavbarDropdown: FC<NavbarDropdownProps> = ({
  children,
  avatar,
}) => {
  return (
    <UncontrolledDropdown nav inNavbar>
      <DropdownToggle
        nav
        role="button"
        data-toggle="dropdown"
        className="nav-link font-weight-bold dropdown-toggle"
      >
        <Avatar {...avatar} />
      </DropdownToggle>
      <DropdownMenu right>{children}</DropdownMenu>
    </UncontrolledDropdown>
  );
};
