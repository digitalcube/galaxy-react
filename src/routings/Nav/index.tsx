export * from './NavLinkItem'
export * from './Nav'
export * from './NavbarBrand'