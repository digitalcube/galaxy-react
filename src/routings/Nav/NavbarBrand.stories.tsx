import React from 'react';
import { Meta, Story } from '@storybook/react';
import { NavbarBrand, NavbarBrandProps } from './NavbarBrand';

const meta: Meta = {
  title: 'Routings/NavbarBrand',
  component: NavbarBrand,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarBrandProps> = (args) => <NavbarBrand {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  name: 'shifter',
  width: 24,
  href: '/',
};
