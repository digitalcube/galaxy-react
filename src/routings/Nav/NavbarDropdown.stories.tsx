import React from 'react';
import { Meta, Story } from '@storybook/react';
import { NavbarDropdown, NavbarDropdownProps } from './NavbarDropdown';
import { Nav } from './Nav';
import { NavLinkItem } from './NavLinkItem';

const meta: Meta = {
  title: 'Routings/NavbarDropdown',
  component: NavbarDropdown,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarDropdownProps> = (args) => (
  <NavbarDropdown {...args} />
);

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});
Default.args = {
  children: 'item',
  avatar: {
    email: 'test@example.com',
    size: 32,
    rounded: true,
  },
};

const IntegratedTemplate: Story<NavbarDropdownProps> = (args) => (
  <nav className="navbar navbar-expand-lg">
    <Nav>
      <NavLinkItem to="#">Guides</NavLinkItem>
      <NavbarDropdown {...args} />
    </Nav>
  </nav>
);
export const InNavbar = IntegratedTemplate.bind({});
InNavbar.args = {
  avatar: {
    email: 'test@example.com',
  },
};
