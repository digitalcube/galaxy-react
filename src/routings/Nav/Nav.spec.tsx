import React from 'react';
import { render } from '@testing-library/react'
import { Default as Nav } from './Nav.stories';

describe('Nav', () => {
  it('to match snapshot', () => {
    const { asFragment } = render(<Nav {...{
        ...Nav.args,
    }}/>);
    expect(asFragment()).toMatchSnapshot()
  });
});
