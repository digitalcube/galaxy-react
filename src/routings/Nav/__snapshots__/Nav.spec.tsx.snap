// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav to match snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg"
  >
    <button
      aria-label="Toggle navigation"
      class="navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse flex-grow-0 collapse"
    >
      <ul
        class="ml-auto navbar-nav nav"
      >
        item
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="/"
          >
            Menu
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="/"
          >
            Menu
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="rounded font-weight-bold nav-link"
            to="/"
          >
            Menu
          </a>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
